lib/net/http/header.rb

```diff
@@ -1,16 +1,159 @@
+#   req['accept'] # => "application/json"
+#   req['ACCEPT'] = 'text/plain'
+#   req['accept'] # => "text/plain"
+#
+# === Field Values
+#
+# A field value may be returned as an array or as a string:
+#
+# - These methods return field values as arrays:
+#
+#   - #get_fields: Returns the array value for the given key,
+#     or +nil+ if it does not exist.
+#   - #to_hash: Returns a hash of all header fields:
+#     each key is a field name; its value is the array value for the field.
+#
+# - These methods return field values as string;
+#   the string value for a field is equivalent to
+#   <tt>self[key.downcase.to_s].join(', '))</tt>:
+#
+#   - #[]: Returns the string value for the given key,
+#     or +nil+ if it does not exist.
+#   - #fetch: Like #[], but accepts a default value
+#     to be returned if the key does not exist.
+#
+# The field value may be set:
+#
+# - #[]=: Sets the value for the given key;
+#   the given value may be a string, a symbol, an array, or a hash.
+# - #add_field: Adds a given value to a value for the given key
+#   (not overwriting the existing value).
+# - #delete: Deletes the field for the given key.
+#
+# Example field values:
+#
+# - \String:
+#
+#     req[:foo] = 'bar'
+#     req[:foo]            # => "bar"
+#     req.get_fields(:foo) # => ["bar"]
+#
+# - \Symbol:
+#
+#     req[:foo] = :bar
+#     req[:foo]            # => "bar"
+#     req.get_fields(:foo) # => ["bar"]
+#
+# - Simple array:
+#
+#     req[:foo] = %w[bar baz bat]
+#     req[:foo]            # => "bar, baz, bat"
+#     req.get_fields(:foo) # => ["bar", "baz", "bat"]
+#
+# - Simple hash:
+#
+#     req[:foo] = {bar: 0, baz: 1, bat: 2}
+#     req[:foo]            # => "bar, 0, baz, 1, bat, 2"
+#     req.get_fields(:foo) # => ["bar", "0", "baz", "1", "bat", "2"]
+#
+# - Nested:
+#
+#     req[:foo] = [%w[bar baz], {bat: 0, bam: 1}]
+#     req[:foo]            # => "bar, baz, bat, 0, bam, 1"
+#     req.get_fields(:foo) # => ["bar", "baz", "bat", "0", "bam", "1"]
+#
+#     req[:foo] = {bar: %w[baz bat], bam: {bah: 0, bad: 1}}
+#     req[:foo]            # => "bar, baz, bat, bam, bah, 0, bad, 1"
+#     req.get_fields(:foo) # => ["bar", "baz", "bat", "bam", "bah", "0", "bad", "1"]
+#
+# == Convenience Methods
+#
+# Various convenience methods retrieve values, set values, query values,
+# set form values, or iterate over fields.
+#
+# === Getters
+#
+# - #[]: Returns the string value for the given field.
+# - #content_length: Returns the integer value of field +:content-length+.
+# - #content_range: Returns the Range value of field +:content-range+.
+# - #content_type: Returns the string value of field +:content-type+.
+# - #main_type: Returns first part of the string value of field +:content-type+.
+# - #sub_type: Returns second part of the string value of field +:content-type+.
+# - #range: Returns an array of Range objects, or +nil+.
+# - #range_length: Returns the integer length of the range given in field +:range+.
```

Comment:
Looking at the documentation and source code, it looks at the `Content-Range` header (and not `Range`).

```suggestion
# - #range_length: Returns the integer length of the range given in field +'Content-Range'+.
```

Reply:
Fixed.